src/vs/base/common/lifecycle.ts

```diff
@@ -430,6 +430,34 @@ export class DisposableStore implements IDisposable {
 
 		return o;
 	}
+
+	/**
+	 * Deletes a disposable from store and disposes of it. This will not throw or warn and proceed to dispose the
+	 * disposable even when the disposable is not part in the store.
+	 */
+	public delete<T extends IDisposable>(o: T): void {
+		if (!o) {
```

Comment:
Type checks should prevent this from ever happening:

![Code_-_Insiders_2h0lrwsH89](https://github.com/microsoft/vscode/assets/2931520/6551dcb2-8dd0-46cb-91a5-3ec9c8178675)

I would either change it to `o: T | undefined` or remove that check (otherwise it might swallow fishy runtime behavior).

Reply:
It aligns with `Set.delete`, the `if (!o)` check if to make it work the same as `DisposableStore.add`